### ISSUE TITLE
problem with less file

### ISSUE THREAD

**Issue by dejour** [prettier/prettier#1967]
before:
  `.siteCount {
     &:extend(.center);
   }
 `
after:
 `
 .siteCount {
    &:  extend(.center);
 }
`
this cause following error;
![image](https://cloud.githubusercontent.com/assets/7224044/26767980/26f21f7c-49d7-11e7-90a2-aebb8b1d0c8c.png)

**Comment by jfranty** [comment]
Using a stand-along `.less` containing the text above prettier 1.4.2 doesn't inject a space for me. However, I've cut the example below down to reliably reproduce this issue for me, yet removing just about any line fixes the issue :

```
$ cat example.less
.foobar {
  &:extend(.foo);
}

.thing {
  &:hover {
    background-color: blue;
    .thing-child {
    }
  }
}

$ ./node_modules/.bin/prettier example.less
.foobar {
  &: extend(.foo);
}

.thing {
  &:hover {
    background-color: blue;
    .thing-child {
    }
  }
}

$ ./node_modules/.bin/prettier --version
1.4.2
```

**Comment by vjeux** [comment]
When it is parsing as scss, it breaks. When it is parsing as less, it works.

We use a heuristic (rough regex) to decide if we should parse things as scss or less. This is why you have to come up with some complex expression to trigger it.

**Comment by vjeux** [comment]
I just published 1.4.3 which fixes it. Thanks for reporting it!

**Comment by jfranty** [comment]
Confirming that my longer .less files are now working after bumping to 1.4.4. Thanks!
